lib/src/native/realm_core.dart

```diff
@@ -1997,56 +1997,48 @@ void _intoRealmValue(Object? value, Pointer<realm_value_t> realm_value, Allocato
     realm_value.ref.values.link.target_table = link.classKey;
     realm_value.ref.type = realm_value_type.RLM_TYPE_LINK;
   } else {
-    switch (value.runtimeType) {
-      case int:
-        realm_value.ref.values.integer = value as int;
-        realm_value.ref.type = realm_value_type.RLM_TYPE_INT;
-        break;
-      case bool:
-        realm_value.ref.values.boolean = value as bool;
-        realm_value.ref.type = realm_value_type.RLM_TYPE_BOOL;
-        break;
-      case String:
-        String string = value as String;
-        final units = utf8.encode(string);
-        final result = allocator<Uint8>(units.length);
-        final Uint8List nativeString = result.asTypedList(units.length);
-        nativeString.setAll(0, units);
-        realm_value.ref.values.string.data = result.cast();
-        realm_value.ref.values.string.size = units.length;
-        realm_value.ref.type = realm_value_type.RLM_TYPE_STRING;
-        break;
-      case double:
-        realm_value.ref.values.dnum = value as double;
-        realm_value.ref.type = realm_value_type.RLM_TYPE_DOUBLE;
-        break;
-      case ObjectId:
-        final bytes = (value as ObjectId).bytes;
-        for (var i = 0; i < 12; i++) {
-          realm_value.ref.values.object_id.bytes[i] = bytes[i];
-        }
-        realm_value.ref.type = realm_value_type.RLM_TYPE_OBJECT_ID;
-        break;
-      case Uuid:
-        final bytes = (value as Uuid).bytes.asUint8List();
-        for (var i = 0; i < 16; i++) {
-          realm_value.ref.values.uuid.bytes[i] = bytes[i];
-        }
-        realm_value.ref.type = realm_value_type.RLM_TYPE_UUID;
-        break;
-      case DateTime:
-        final microseconds = (value as DateTime).toUtc().microsecondsSinceEpoch;
-        final seconds = microseconds ~/ _microsecondsPerSecond;
-        int nanoseconds = _nanosecondsPerMicrosecond * (microseconds % _microsecondsPerSecond);
-        if (microseconds < 0 && nanoseconds != 0) {
-          nanoseconds = nanoseconds - _nanosecondsPerMicrosecond * _microsecondsPerSecond;
-        }
-        realm_value.ref.values.timestamp.seconds = seconds;
-        realm_value.ref.values.timestamp.nanoseconds = nanoseconds;
-        realm_value.ref.type = realm_value_type.RLM_TYPE_TIMESTAMP;
-        break;
-      default:
-        throw RealmException("Property type ${value.runtimeType} not supported");
+    if (value is int) {
```

Comment:
The code will work but why are we doing this change. Did the previous version had drawbacks?

Reply:
ok I understand now